base/mpfr.jl

```diff
@@ -765,12 +765,15 @@ function with_bigfloat_precision(f::Function, precision::Integer)
 end
 
 function string(x::BigFloat)
-    lng::Int32 = 127
+    # In general, the number of decimal places needed to read back the number exactly is
+    # ceil(log(10, 2^precision(x))) or ceil(precision(x) * log10(2))
+    dp = ceil(Int32, precision(x) * 0.3010299956639812)
```

Comment:
does this need to be `log10(big(2))`, or does the Float64 approximation work for all reasonable values of BigFloat that could be encountered here